irods/access.py

```diff
@@ -45,10 +63,31 @@ def __init__(self, access_name, path, user_name='', user_zone='', user_type=None
         self.user_zone = user_zone
         self.user_type = user_type
 
+    def copy(self, decanonicalize = False):
+        other = copy.deepcopy(self)
+        if decanonicalize:
+            replacement_string = { 'read object':'read',
+                                   'read_object':'read',
+                                   'modify object':'write',
+                                   'modify_object':'write'}.get(self.access_name)
+            other.access_name = replacement_string if replacement_string is not None \
+                                else self.access_name
+        return other
+
     def __repr__(self):
         object_dict = vars(self)
         access_name = self.access_name.replace(' ','_')
         user_type_hint = ("({user_type})" if object_dict.get('user_type') is not None else "").format(**object_dict)
         return "<iRODSAccess {0} {path} {user_name}{1} {user_zone}>".format(access_name,
                                                                             user_type_hint,
                                                                             **object_dict)
+
+class _iRODSAccess_4_2(iRODSAccess):
```

Comment:
I feel like it may be better to avoid mentioning 4.2.x specifically here as this has been the historical access model (from way back). Perhaps "legacy" or pre_4_3 or something? I don't feel very strongly about this, but it struck me as odd that it's mentioning 4.2.x and applies to everything before 4.3.x.

Reply:
same feeling here...  pre_430 seems a tad more on the nose...

Reply:
Ok .  Seems good.